crates/noirc_evaluator/src/object/array.rs

```diff
@@ -151,7 +151,7 @@ impl Array {
                 .expect("ice: arrays must have at least one element in them"),
         );
 
-        for pred in predicates_iter.next() {
+        if let Some(pred) = predicates_iter.next() {
```

Comment:
Thanks for catching, this was unintended, as the compiler should multiply all predicates together.

`for pred in predicates_iter {}` or ` while let Some(pred) = predicate_iter.next() {}`